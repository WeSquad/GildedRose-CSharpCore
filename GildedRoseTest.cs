using Xunit;
using System.Collections.Generic;
using ApprovalTests.Reporters;
using ApprovalTests;
using System;

namespace csharpcore
{
    [UseReporter(typeof(DiffReporter))]
    public class GildedRoseTest
    {
        [Fact]
        public void FooItemTest()
        {
            IList<Item> Items = new List<Item> { new Item { Name = "foo", SellIn = 0, Quality = 0 } };
            GildedRose app = new GildedRose(Items);
            app.UpdateQuality();
            Assert.Equal("foo", Items[0].Name);
        }

        [Fact]
        public void TestUseOfApprovalTests()
        {
            var names = new[] { "Llewellyn", "James", "Dan", "Jason", "Katrina" };
            Array.Sort(names);
            Approvals.VerifyAll(names, "");
        }
    }
}