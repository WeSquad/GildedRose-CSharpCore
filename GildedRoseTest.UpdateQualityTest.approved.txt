[foo,0,0] => name: foo, sell in: -1, quality: 0
[foo,0,1] => name: foo, sell in: -1, quality: 0
[foo,0,-2] => name: foo, sell in: -1, quality: -2
[foo,0,2] => name: foo, sell in: -1, quality: 0
[foo,-1,0] => name: foo, sell in: -2, quality: 0
[foo,-1,1] => name: foo, sell in: -2, quality: 0
[foo,-1,-2] => name: foo, sell in: -2, quality: -2
[foo,-1,2] => name: foo, sell in: -2, quality: 0
[Aged Brie,0,0] => name: Aged Brie, sell in: -1, quality: 2
[Aged Brie,0,1] => name: Aged Brie, sell in: -1, quality: 3
[Aged Brie,0,-2] => name: Aged Brie, sell in: -1, quality: 0
[Aged Brie,0,2] => name: Aged Brie, sell in: -1, quality: 4
[Aged Brie,-1,0] => name: Aged Brie, sell in: -2, quality: 2
[Aged Brie,-1,1] => name: Aged Brie, sell in: -2, quality: 3
[Aged Brie,-1,-2] => name: Aged Brie, sell in: -2, quality: 0
[Aged Brie,-1,2] => name: Aged Brie, sell in: -2, quality: 4
[Backstage passes to a TAFKAL80ETC concert,0,0] => name: Backstage passes to a TAFKAL80ETC concert, sell in: -1, quality: 0
[Backstage passes to a TAFKAL80ETC concert,0,1] => name: Backstage passes to a TAFKAL80ETC concert, sell in: -1, quality: 0
[Backstage passes to a TAFKAL80ETC concert,0,-2] => name: Backstage passes to a TAFKAL80ETC concert, sell in: -1, quality: 0
[Backstage passes to a TAFKAL80ETC concert,0,2] => name: Backstage passes to a TAFKAL80ETC concert, sell in: -1, quality: 0
[Backstage passes to a TAFKAL80ETC concert,-1,0] => name: Backstage passes to a TAFKAL80ETC concert, sell in: -2, quality: 0
[Backstage passes to a TAFKAL80ETC concert,-1,1] => name: Backstage passes to a TAFKAL80ETC concert, sell in: -2, quality: 0
[Backstage passes to a TAFKAL80ETC concert,-1,-2] => name: Backstage passes to a TAFKAL80ETC concert, sell in: -2, quality: 0
[Backstage passes to a TAFKAL80ETC concert,-1,2] => name: Backstage passes to a TAFKAL80ETC concert, sell in: -2, quality: 0
